perfkitbenchmarker/linux_benchmarks/ping_benchmark.py

```diff
@@ -62,17 +62,41 @@ def Run(benchmark_spec):
     A list of sample.Sample objects.
   """
   vms = benchmark_spec.vms
-  if not vms[0].IsReachable(vms[1]):
-    logging.warn('%s is not reachable from %s', vms[1], vms[0])
+  results = []
+  for origin_index in [0, 1]:
+    origin_vm = vms[origin_index]
+    receiving_vm = vms[origin_index ^ 1]
+    results = results + _RunPing(origin_vm,
+                                 receiving_vm,
+                                 receiving_vm.internal_ip,
```

Comment:
Allowing `ping` using external ips requires changes to the firewall (at least for gcp), and since ping uses ICMP, it didn't see how it could be done with `vm.AllowPort()`.  This makes a supporting external ips for ping a much bigger piece of work--either it requires changing the firewall for all benchmarks and (potentially) all providers, or adding some way to change the firewall specifically for the ping benchmark.